config/proxy_config.go

```diff
@@ -0,0 +1,120 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/Layr-Labs/eigenda-proxy/common"
+	"github.com/Layr-Labs/eigenda-proxy/config/eigendaflags"
+	eigendaflags_v2 "github.com/Layr-Labs/eigenda-proxy/config/eigendaflags/v2"
+	"github.com/Layr-Labs/eigenda-proxy/server"
+	"github.com/Layr-Labs/eigenda-proxy/store"
+	"github.com/Layr-Labs/eigenda-proxy/store/generated_key/memstore"
+	"github.com/Layr-Labs/eigenda-proxy/store/generated_key/memstore/memconfig"
+	"github.com/Layr-Labs/eigenda-proxy/verify/v1"
+	"github.com/urfave/cli/v2"
+
+	"github.com/Layr-Labs/eigenda/api/clients"
+)
+
+// ProxyConfig ... Higher order config which bundles all configs for orchestrating
+// the proxy server with necessary client context
+type ProxyConfig struct {
+	ServerConfig        server.Config
+	EdaClientConfigV1   clients.EigenDAClientConfig
+	EdaVerifierConfigV1 verify.Config
+
+	EdaClientConfigV2 common.ClientConfigV2
+
+	MemstoreConfig *memconfig.SafeConfig
+	StorageConfig  store.Config
+
+	// Enabling will turn on memstore for both V1 && V2
+	//
+	// Currently the memstore is not persisted to disk, and is lost when proxy is stopped. Testing migrations is
+	// currently not supported because it requires restarting the proxy with new flags, which will cause the
+	// memstore db to be wiped.
+	//
+	MemstoreEnabled bool
+
+	EigenDAV2Enabled bool
+
+	PutRetries       uint
+	MaxBlobSizeBytes uint
+
+	EigenDACertVerifierAddress string
```

Comment:
why are these here and not part of other lower-level configs? Is it because they are shared between a bunch of different services?

Reply:
Fair point. In [9f73ebe](https://github.com/Layr-Labs/eigenda-proxy/pull/321/commits/9f73ebe25de836b7dc7aeb3e4ef2d0eaec2b682e) I moved `EigenDACertVerifierAddress ` into a config struct (that's where I should have put it in the first place).

As for `PutRetries` and `MaxBlobSyzeBytes`... I think it makes sense to just leave them where they are for now (this is how they were configured prior to this PR). It's not clear to me upon initial inspection where these would best be moved to.